subprojects/core/src/integTest/groovy/org/gradle/api/internal/project/LifecycleAwareProjectIntegrationTest.groovy

```diff
@@ -87,4 +87,39 @@ class LifecycleAwareProjectIntegrationTest extends AbstractIntegrationSpec {
         then:
         outputContains "a foo=bar1"
     }
+
+    def 'LifecycleAwareProject wrappers are symmetrically equals to wrapped projects and to each other'() {
+        given:
+        settingsFile """
+            include(":a")
+        """
+
+        buildFile """
+            ext["wrappedBySubprojects"] = subprojects.toList().get(0)
```

Comment:
This test is IP-unsafe, but we can make it safe by collecting the project instances in a build service. This is valid use of the public API.

Collecting project instances in a shared data-structure is not a violation of IP by itself, as long as project's mutable state is not touched. So we should test this with and without IP.

Reply:
`LifecycleAwareProjectIntegrationTest` entirely is not IP safe and that's why it `@Requires(IntegTestPreconditions.NotIsolatedProjects)`. This test exercises wrappers behavior mostly, and not the correctness of IP violations reporting by them

I think it would be good to have some similar coverage  for IP world and `ProblemReportingProject`

Reply:
This test should not be in a test class with IP precondition to begin with. It should be executed by all our standard runners, including the one with IP -- in effect, testing all the wrappers.

Reply:
Created a test that IP compatible and exercises different combinations of projects for equality check